### PR TITLE
folly: 2023.02.27.00 -> 2024.01.08.00

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

## Description of changes
Diff: https://github.com/facebook/folly/compare/v2023.02.27.00...v2024.01.08.00

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/folly/default.nix [hunk 1]
```diff
@@ -23,13 +23,13 @@
 
 stdenv.mkDerivation rec {
   pname = "folly";
-  version = "2023.02.27.00";
+  version = "2023.10.02.00";
```
Comment (FliegendeWurst) [inline pkgs/development/libraries/folly/default.nix:26]: 2023.10.30 has been tagged by now.

Comment (wegank) [reply pkgs/development/libraries/folly/default.nix:26]: Also, 2024.01.01.00 is out.

Comment (mfrw) [reply pkgs/development/libraries/folly/default.nix]: Upgraded to: 2024.01.01.00
/cc @wegank @FliegendeWurst

**Review by FliegendeWurst — COMMENTED** [review]
Package builds successfully on x64-linux :+1: 

However, `nixpkgs-review pr --print-result 258942` fails with an error message related to:

`error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "filemagic" at /tmp/pr-258942/nixpkgs/pkgs/development/python-modules/jira/default.nix:5'`

It doesn't look like the issue stems from this PR though.

File: pkgs/development/libraries/folly/default.nix [hunk 2]
```diff
@@ -22,13 +22,13 @@
...
   pname = "folly";
-  version = "2023.02.27.00";
+  version = "2024.01.08.00";
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "folly";
     rev = "v${version}";
-    sha256 = "sha256-DfZiVxncpKSPn9BN25d8o0/tC27+HhSG/t53WgzAT/s=";
+    sha256 = "sha256-s2havLUpmnYP6shCDtcqblGTqFq1nGsfHWd8b8A0t7Y=";
```
Comment (wegank) [inline pkgs/development/libraries/folly/default.nix:31]:
```suggestion
    hash = "sha256-s2havLUpmnYP6shCDtcqblGTqFq1nGsfHWd8b8A0t7Y=";
```

**Comment by wegank** [comment]
Is setting `CMAKE_LIBRARY_ARCHITECTURE` still necessary?

**Comment by wegank** [comment]
This also breaks `folly` on x86_64-darwin (tested on master), so @NixOS/darwin-maintainers?

**Comment by viraptor** [comment]
> error: building of '/nix/store/v24bxnbqhaa03d21y34yxq4sr7dgsr2x-llvm-16.0.6.drv^lib,out' from .drv file timed out after 3600 seconds

I don't think that's unexpected for llvm to take a long time, but we're rebuilding it... because of staging? The build works just fine for me when rebased on master.

The architecture can probably go since https://github.com/facebook/folly/commit/0e753e5bd1780208173e832181faf205d093cc4a was merged.

Will this cache llvm? 🤔 
@ofborg build llvm_16

**Comment by wegank** [comment]
Done in #285345.

**Pull request closed without merging**